src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(anotherAnimal) {
+    if (anotherAnimal instanceof Herbivore && !anotherAnimal.hidden) {
+      anotherAnimal.health -= 50;
+    }
+
+    if (anotherAnimal.health <= 0) {
+      Animal.alive = Animal.alive.filter((enimal) => enimal.health > 0);
+    }
```

Comment:
this if statement needs to be checked only if previous one is correct, so if animal will be an Carnivore or hidden Herbivore, it wont even check it

Reply:
nice, thanks you!